doc/manual/src/advanced-topics/post-build-hook.md

```diff
@@ -33,12 +33,18 @@ distribute the public key for verifying the authenticity of the paths.
 example-nix-cache-1:1/cKDz3QCCOmwcztD2eV6Coggp6rqc9DGjWv7C0G+rM=
 ```
 
-Then, add the public key and the cache URL to your `nix.conf`'s
+Then on any machine that will consume from or upload to the cache, add
+the public key and the cache URL to your `nix.conf`'s
 `trusted-public-keys` and `substituters` options:
 
     substituters = https://cache.nixos.org/ s3://example-nix-cache
     trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= example-nix-cache-1:1/cKDz3QCCOmwcztD2eV6Coggp6rqc9DGjWv7C0G+rM=
 
+Machines which build packages for the cache will also need the path to
+the key file added to their `nix.conf`:
+
+    secret-key-files = /etc/nix/key.private
+
```

Comment:
These are instructions which correctly started as imperative. No reason to switch to descriptive phrasing.

```suggestion
On machines which build packages for the cache, update [`nix.conf`] with the path to their key file ([`secret-key-files`]):

[`secret-key-files`]: ../command-ref/conf-file.md#secret-key-files

    secret-key-files = /etc/nix/key.private

```